odxtools/comparam_subset.py

```diff
@@ -115,11 +115,10 @@ def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
 class Comparam(BaseComparam):
     dop_ref: OdxLinkRef
     physical_default_value: Optional[str] = field(default=None, init=False)
-    _dop: Optional[DataObjectProperty] = field(default=None, init=False)
 
     @staticmethod
-    def from_et(et_element, doc_frags: List[OdxDocFragment]) -> ComplexComparam:
-        result = Comparam.__new__(ComplexComparam)
+    def from_et(et_element, doc_frags: List[OdxDocFragment]) -> "Comparam":
+        result = Comparam.__new__(Comparam)
```

Comment:
why not just `result = Comparam()`?

Reply:
because then the "offical" constructor would need to be called and for this I need all attributes stored by the `Comparam` as well as the `BaseComparam` classes. (the problem with the `from_et()` idiom is that it does not work very well with inheritance, thus the split of the instantiation of such classes into object creation (`XYZ.__new__()`) and their actual initialization (`__init_from_et__()`.) If you know a better solution, I'd be happy to give it a shot...

Reply:
since it's existing code, I think it would suffice to explain the magic behind __new__ usage in a comment

Reply:
It's still pretty new code (#104 from last December). I suppose I should have put this into a separate PR instead of burying it in #104. (When I separated the original monster PR for that, I forgot about that detail. Sorry for that...)

Reply:
BTW: In the medium term, I intend to refactor the class hierarchy a bit so that it resembles the XSD more closely. this means that this pattern will see quite a bit more use as the XSD makes quite a bit of use of inheritance...

Reply:
ok, I've added some comments. I hope that I don't forget to add them to future occurrences of this pattern...